### ISSUE TITLE
bc connectToChain, the register-node call and error handling

### ISSUE THREAD

**Issue by phpai** [openkfw/TruBudget#19]
In GitLab by @kevinbader on Jun 19, 2018, 09:57

- If `multichaind` dies unexpectedly (exit code `!= 1`), `multichaind` is not restarted and it says `Connect: Success!`, which is obviously wrong.
- If the node has already been registered in an earlier run, `multichaind` connects right away and `address` is not set, causing `registerNodeAtMaster` to busy-wait.

Related: #90

**Comment by phpai** [comment]
In GitLab by @kevinbader on Jun 19, 2018, 10:04

changed title from **bc connectToChain{- and the after-the-fact register-node call-}** to **bc connectToChain{+, the register-node call and error handling+}**

**Comment by phpai** [comment]
In GitLab by @kevinbader on Jun 19, 2018, 10:04

changed the description

**Comment by phpai** [comment]
In GitLab by @Stezido on Nov 16, 2018, 12:46

#151

**Comment by phpai** [comment]
In GitLab by @Stezido on Nov 16, 2018, 14:45

created branch [`103-fix-endless-loop-of-restarted-slave`](https://innersource.stackato.de/trubudget/trubudget-app/compare/master...103-fix-endless-loop-of-restarted-slave)

**Comment by phpai** [comment]
In GitLab by @Stezido on Nov 16, 2018, 14:45

mentioned in merge request !125

**Comment by Stezido** [comment]
```
If the node has already been registered in an earlier run, multichaind connects right away and
address is not set, causing registerNodeAtMaster to busy-wait.
```
### Description
This behavior appears when updating a slave node to a newer version. The address of the newly created slave is a new one, so the master node cannot identify the new updated slave node as the old one. The master node handles the request as if the slave node is a new node.

 ### Solution
The new updated slave node has to somehow use the old wallet address by using the same private key
Maybe it is possible via dumpprivkey and importprivkey or the wallet.dat file has to be copied.
#### Sources:
 (https://www.multichain.com/qa/4207/connect-blockchain-using-specific-wallet-address-different)
(https://www.multichain.com/qa/10357/using-a-external-address-in-a-existing-wallet)

**Comment by mayrmartin** [comment]
Solved in #515
